### PR TITLE
[lldb] Additional pieces towards OpenBSD support

### PR DESCRIPTION
**Pull request by brad0**

### CODE REVIEW

File: lldb/source/Core/FormatEntity.cpp [hunk 1]
```diff
@@ -1245,9 +1245,10 @@ bool FormatEntity::Format(const Entry &entry, Stream &s,
             llvm::Triple::OSType ostype = arch.IsValid()
                                               ? arch.GetTriple().getOS()
                                               : llvm::Triple::UnknownOS;
-            if ((ostype == llvm::Triple::FreeBSD) ||
-                (ostype == llvm::Triple::Linux) ||
-                (ostype == llvm::Triple::NetBSD)) {
+            if (ostype == llvm::Triple::Linux ||
```
Comment (emaste) [inline lldb/source/Core/FormatEntity.cpp:1248]: OSes were in alpha order before -- it doesn't matter too much but both consistency across the code base and avoiding churn are good goals. Is the order is generally Linux then BSDs in alpha order across the tree?

Comment (brad0) [reply lldb/source/Core/FormatEntity.cpp]: There are some cases like that in some areas. I try to keep the BSD's grouped together and in order of DragonFly, FreeBSD, NetBSD, OpenBSD where relevant. I took out that part.

File: lldb/test/API/tools/lldb-server/thread-name/main.cpp [hunk 2]
```diff
@@ -1,4 +1,7 @@
 #include <pthread.h>
+#ifdef __OpenBSD__
```
Comment (bulbazord) [inline lldb/test/API/tools/lldb-server/thread-name/main.cpp:2]: For consistency, can you change this to `if defined(__OpenBSD__)`? Not a big deal though

**Review by bulbazord — APPROVED** [review]
LGTM overall, small nit

File: lldb/test/API/tools/lldb-server/thread-name/main.cpp (hunk 2 above)
Comment (brad0) [reply lldb/test/API/tools/lldb-server/thread-name/main.cpp:2]: Sure.

Comment (brad0) [reply lldb/test/API/tools/lldb-server/thread-name/main.cpp]: Done.

**Review by emaste — APPROVED** [review]

**Pull request merged**